src/librbd/ObjectMap.cc

```diff
@@ -110,6 +110,32 @@ bool ObjectMap<I>::object_may_exist(uint64_t object_no) const
   return exists;
 }
 
+template <typename I>
+bool ObjectMap<I>::object_may_not_exist(uint64_t object_no) const
+{
+  ceph_assert(m_image_ctx.snap_lock.is_locked());
+
+  // Fall back to default logic if object map is disabled or invalid
+  if (!m_image_ctx.test_features(RBD_FEATURE_OBJECT_MAP,
+                                 m_image_ctx.snap_lock)) {
+    return true;
```

Comment:
Nit: should this be ```false``` since we don't know if it exists or not? Same w/ below.

Reply:
... eh, never mind. I see how it's being used below.

Reply:
May be the method name `object_may_not_exist` is confusing and this could be improved? To me it would be more clear if it were `object_may_unexist`, but I don't know how English it is.

Reply:
I don't have a better suggestion (```unexists``` isn't a word) -- I think it's fine